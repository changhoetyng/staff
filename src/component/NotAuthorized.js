import React from 'react';

const NotFound = () => {
  return <h1>400 You are not authorized</h1>;
}

export default NotFound;