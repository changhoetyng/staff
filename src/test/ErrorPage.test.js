import React from "react";
import { configure, shallow } from "enzyme";

// Import Page
import NotAuthorized from "../component/NotAuthorized";
import NotFound from "../component/NotFound";

describe("Render Error Page", () => {
  const notAuthorized = shallow(<NotAuthorized />);
  const notFound = shallow(<NotFound />);

  test("Render not authorized page", () => {
    expect(notAuthorized.find("h1").text()).toContain(
      "400 You are not authorized"
    );
  });

  test("Render not found page", () => {
    expect(notFound.find("h1").text()).toContain("404 Page Not Found");
  });

  test("Snapshot Not Authorized page", () => {
    expect(notAuthorized).toMatchSnapshot();
  });
  test("Snapshot Not Found page", () => {
    expect(notFound).toMatchSnapshot();
  });
});
