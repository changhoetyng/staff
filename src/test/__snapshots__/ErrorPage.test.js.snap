// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Error Page Snapshot Not Authorized page 1`] = `
<h1>
  400 You are not authorized
</h1>
`;

exports[`Render Error Page Snapshot Not Found page 1`] = `
<h1>
  404 Page Not Found
</h1>
`;
